### PR TITLE
wrap slice_combine() errors

### PR DESCRIPTION
**Pull request by romainfrancois**
``` r
library(dplyr, warn.conflicts = FALSE)
df <- data.frame(x = 1:2)
gdf <- group_by(df, x)

slice(gdf, matrix(c(1, 2), ncol = 2))
#> Error in `slice()`:
#>   Problem while checking indices.
#>   ℹ The error occurred in group 1: x = 1.
#> Caused by error in `vec_cast()`:
#>   Can't convert <integer[,2]> to <integer>.
#>   Cannot decrease dimensions.
slice(gdf, "a")
#> Error in `slice()`:
#>   Problem while checking indices.
#>   ℹ The error occurred in group 1: x = 1.
#> Caused by error:
#>   Invalid result of type <character>.
#>   ℹ Indices must be positive or negative integers.
slice(gdf, c(1, -1))
#> Error in `slice()`:
#>   Problem while checking indices.
#>   ℹ The error occurred in group 1: x = 1.
#> Caused by error:
#>   Indices must be all positive or all negative.
#>   ℹ Got 1 positives, 1 negatives.
```

<sup>Created on 2021-12-10 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1.9000)</sup>

### CODE REVIEW

File: R/slice.R [hunk 1]
```diff
@@ -323,47 +323,54 @@ slice_bullets <- function(cnd, error_call, index) {
...
+      current_rows <- rows[[group]]
+      res <- chunks[[group]]
+
+      if (is.logical(res) && all(is.na(res))) {
+        res <- integer()
+      } else if (is.numeric(res)) {
+        if (is.matrix(res) && ncol(res) == 1) {
+          res <- as.vector(res)
+        }
+        res <- fix_call(vec_cast(res, integer()), NULL)
```
Comment (romainfrancois) [inline R/slice.R:338]: Perhaps this `fix_call()` is not useful. Without it we currently would get: 

``` r
library(dplyr, warn.conflicts = FALSE)
df <- data.frame(x = 1:2)
gdf <- group_by(df, x)

slice(gdf, matrix(c(1, 2), ncol = 2))
#> Error in `slice()`:
#>   Problem while checking indices.
#>   ℹ The error occurred in group 1: x = 1.
#> Caused by error in `stop_vctrs()`:
#>   Can't convert <integer[,2]> to <integer>.
#>   Cannot decrease dimensions.
```

<sup>Created on 2021-12-10 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1.9000)</sup>

but supposedly `vec_cast()` with a future version of `vctrs` ?

Comment (lionel-) [reply R/slice.R]: With future vctrs you'll need to supply a `NULL` call explicitly, which the `fix_call()` currently emulates.

File: R/slice.R [hunk 2]
```diff
@@ -323,47 +323,54 @@ slice_bullets <- function(cnd, error_call, index) {
...
+          i = glue("Got {n_positive} positives, {n_negative} negatives.")
+        )
+        abort(bullets, call = NULL)
+      }
+
+      slice_indices[[group]] <- current_rows[res]
+    }, error = function(cnd) {
+    mask$set_current_group(group)
+    bullets <- c(
+      "Problem while checking indices.",
```
Comment (lionel-) [inline R/slice.R:368]: Is it useful to make the distinction between computing and checking, from the user point of view? Perhaps it's better to always use "computing" for simplicity.

File: tests/testthat/_snaps/slice.md [hunk 3]
```diff
@@ -3,21 +3,60 @@
     Code
       (expect_error(slice(df, matrix(c(1, 2), ncol = 2))))
     Output
-      <error/vctrs_error_incompatible_type>
-      Error in `slice()`: Can't convert <integer[,2]> to <integer>.
-      Cannot decrease dimensions.
+      <error/rlang_error>
+      Error in `slice()`:
+        Problem while checking indices.
+      Caused by error in `vec_cast()`:
```
Comment (lionel-) [inline tests/testthat/_snaps/slice.md:9]: Needs a `fix_call(call = NULL)`?

Comment (romainfrancois) [reply tests/testthat/_snaps/slice.md:9]: That's what I had originally, but then I assumed that without the `fix_call()` we would eventually get Caused by error in `vec_cast()`. 

I can make it NULL

**Pull request merged**